docs/source/export.rst

```diff
@@ -1,10 +1,462 @@
+  called the Frame Evaluation API to safely trace PyTorch graphs. This
+  provides a massively improved graph capturing experience, with much fewer
+  rewrites needed in order to fully trace the PyTorch code.
+
+* **AOT Autograd** provides a functionalized PyTorch graph and ensures the graph
+  is decomposed/lowered to the small defined Core ATen operator set.
+
+* **Torch FX (torch.fx)** is the underlying representation of the graph,
+  allowing flexible Python-based transformations.
+
+
+Existing frameworks
+^^^^^^^^^^^^^^^^^^^
+
+:func:`torch.compile` also utilizes the same PT2 stack as ``torch.export``, but
+is slightly different:
+
+* **JIT vs. AOT**: :func:`torch.compile` is a JIT compiler whereas
+  which is not intended to be used to produce compiled artifacts outside of
+  deployment.
+
+* **Partial vs. Full Graph Capture**: When :func:`torch.compile` runs into an
+  untraceable part of a model, it will "graph break" and fall back to running
+  the program in the eager Python runtime. In comparison, ``torch.export`` aims
+  to get a full graph representation of a PyTorch model, so it will error out
+  when something untraceable is reached. Since ``torch.export`` produces a full
+  graph disjoint from any Python features or runtime, this graph can then be
+  saved, loaded, and run in different environments and languages.
+
+* **Usability tradeoff**: Since :func:`torch.compile` is able to fallback to the
+  Python runtime whenever it reaches something untraceable, it is a lot more
+  flexible. ``torch.export`` will instead require users to provide more
+  information or rewrite their code to make it traceable.
+
+Compared to :func:`torch.fx.symbolic_trace`, ``torch.export`` traces using
+TorchDynamo which operates at the Python bytecode level, giving it the ability
+to trace arbitrary Python constructs not limited by what Python operator
+overloading supports. Additionally, ``torch.export`` keeps fine-grained track of
+tensor metadata, so that conditionals on things like tensor shapes do not
+fail tracing. In general, ``torch.export`` is expected to work on more user
+programs, and produce lower-level graphs (at the ``torch.ops.aten`` operator
+level). Note that users can still use :func:`torch.fx.symbolic_trace` as a
+preprocessing step before ``torch.export``.
+
+Compared to :func:`torch.jit.script`, ``torch.export`` does not capture Python
+control flow or data structures, but it supports more Python language features
+than TorchScript (as it is easier to have comprehensive coverage over Python
+bytecodes). The resulting graphs are simpler and only have straight line control
+flow (except for explicit control flow operators).
+
+Compared to :func:`torch.jit.trace`, ``torch.export`` is sound: it is able to
+trace code that performs integer computation on sizes and records all of the
+side-conditions necessary to show that a particular trace is valid for other
+inputs.
+
```

Comment:
What do you think about move this to somewhere later so that we start with a one-sentence overview and then immediately go into an example?

Reply:
Hmm, if we do that then I'm not sure how to organize the document. 

Should we put both examples (w/o dynamic shape and w/ dynamic shape)?
Where should we include the stuff about specialization? 
Where should the graph invariants be explained then? Right after the example?

Reply:
I mean we start with the most basic example so that users get some intuition of what it does, then we go through detailed explanation and other examples. What do you think?

Reply:
Ok, I added the example and it looks fine. What do you think?

Reply:
Thanks! I mean can we have the most *basic* example, without additional concepts constraints/dynamic dims. Those concepts can be introduced in later sections.

A few other nits about examples:
1) What do you think about splitting export results (the printed exported program part) into a separate code block so that it is clear that is the output
2) Maybe we can manually format the GraphSignature print a bit so that each field is on a new line so that it is easier to read?

Reply:
updated!